internal/discovery/node_discoverer.go

```diff
@@ -0,0 +1,36 @@
+package discovery
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+)
+
+// NodeGetter get nodes with informers.
+type NodeGetter interface {
+	Get(name string) (ret *corev1.Node, err error)
+}
```

Comment:
Since this file contains only contains the getter rather than the discoverer we could just rename it `node_getter.go`.

Reply:
I renamed it to node_lister.go since I changed it to leverage the k8s interface that is "lister"